tests/unit/aiplatform/test_docker_utils.py

```diff
@@ -154,6 +154,112 @@ def test_run_prediction_container_with_all_parameters(
             detach=True,
         )
 
+    @mock.patch.dict(os.environ, {}, clear=True)
+    def test_run_prediction_container_with_envs_replaced_by_envs(
+        self, tmp_path, docker_client_mock
+    ):
+        serving_container_environment_variables = {
+            "VAR_1": "foo",
+            "VAR_2": "$(VAR_3) bar",
+            "VAR_3": "$(VAR_1) bar",
+        }
+        environment = {k: v for k, v in serving_container_environment_variables.items()}
+        environment[prediction.AIP_HTTP_PORT] = prediction.DEFAULT_AIP_HTTP_PORT
+        environment[prediction.AIP_HEALTH_ROUTE] = None
+        environment[prediction.AIP_PREDICT_ROUTE] = None
+        environment[prediction.AIP_STORAGE_URI] = ""
+        environment[
+            run._ADC_ENVIRONMENT_VARIABLE
+        ] = run._DEFAULT_CONTAINER_CRED_KEY_PATH
+        # Envs referencing earlier entries will be changed. Those envs referencing later
+        # entries won't be changed.
+        environment["VAR_3"] = "foo bar"
+
+        run.run_prediction_container(
+            self.IMAGE_URI,
+            serving_container_environment_variables=serving_container_environment_variables,
+        )
+
+        docker_client_mock.containers.run.assert_called_once_with(
+            self.IMAGE_URI,
+            command=None,
+            entrypoint=None,
+            ports={prediction.DEFAULT_AIP_HTTP_PORT: None},
+            environment=environment,
+            volumes=[],
+            detach=True,
+        )
+
+    @mock.patch.dict(os.environ, {}, clear=True)
+    def test_run_prediction_container_with_command_replaced_by_envs(
+        self, tmp_path, docker_client_mock
+    ):
+        serving_container_command = ["$(VAR_1)", "$$(VAR_1)", "$(VAR_2)"]
+        serving_container_environment_variables = {
+            "VAR_1": "foo",
+        }
+        environment = {k: v for k, v in serving_container_environment_variables.items()}
+        environment[prediction.AIP_HTTP_PORT] = prediction.DEFAULT_AIP_HTTP_PORT
+        environment[prediction.AIP_HEALTH_ROUTE] = None
+        environment[prediction.AIP_PREDICT_ROUTE] = None
+        environment[prediction.AIP_STORAGE_URI] = ""
+        environment[
+            run._ADC_ENVIRONMENT_VARIABLE
+        ] = run._DEFAULT_CONTAINER_CRED_KEY_PATH
+        # Command references existing environment variables.
+        expected_entrypoint = ["foo", "$$(VAR_1)", "$(VAR_2)"]
```

Comment:
Can you check to make sure that if the user escapes the syntax, `"$$(VAR_1)"`, that the expected rendering is `"$$(VAR_1)"` and not `"$(VAR_1)"`

Reply:
Should be `"$(VAR_1)` as https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.22/#envvar-v1-core. Thanks!